presto-docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -268,6 +271,40 @@ and a file system location of ``s3://test_bucket/test_schema/test_table``:
         location = 's3://test_bucket/test_schema/test_table')
     )
 
+Caching Support
+----------------
+
+Manifest File Caching
+^^^^^^^^^^^^^^^^^^^^^^
+
+Starting from Iceberg version 1.1.0, Apache Iceberg provides a mechanism to cache the contents of Iceberg manifest files in memory. This manifest caching feature helps
+to reduce repeated reads of small Iceberg manifest files from remote storage.
+
+.. note::
+
+    Currently, manifest file caching is supported for Hadoop and Nessie catalogs in the Presto Iceberg connector.
+
+The following configuration properties are available:
+
+====================================================   =============================================================   ============
+Property Name                                          Description                                                     Default
+====================================================   =============================================================   ============
+``iceberg.io.manifest.cache-enabled``                  Enable/disable the manifest caching feature. This feature       ``false``
+                                                       is only available if ``iceberg.catalog.type`` is ``hadoop``
+                                                       or ``nessie``.
+
+``iceberg.io-impl``                                    Custom FileIO implementation to use in a catalog. It must       ``org.apache.iceberg.hadoop.HadoopFileIO``
```

Comment:
also by introducing this field I would have expected something like 
```
if (ioImpl != null) {
    properties.put(FILE_IO_IMPL, ioImpl);
}
```
in `IcebergResourceFactory`

Reply:
@nastra As I checked in Iceberg core, In case when we query the Hadoop catalog from Presto. here in [HadoopCatalog.initialize()](https://github.com/apache/iceberg/blob/apache-iceberg-1.4.1/core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java#L115) Catalog configuration as not being set via [HadoopFileIO.initialize()](https://github.com/apache/iceberg/blob/apache-iceberg-1.4.1/core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java#L81) but if fileIOImpl config is set explicaity then it creates HadoopFileIO object via [CatalogUtil.loadFileIO()](https://github.com/apache/iceberg/blob/apache-iceberg-1.4.1/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L239) which is also responsible of initialising Catalog configuration (including which are being passed from Presto `io.manifest.cache-enabled`). That is why I had set `org.apache.iceberg.hadoop.HadoopFileIO` explicitely here since Hadoop & Nessie both uses HadoopFileIO.
Please let me know if there is any better way of handling this?

Reply:
thanks for the explanation. The issue that still remains is that there's a risk that a user might configure a different `FileIO` via `iceberg.io-impl` than the underlying catalog uses, which will cause wrong results

Reply:
Exactly, NessieCatalog takes care of initializing it. So it's not an issue there. Thanks for conforming. 
If needed, I can raise a PR for initialization of FileIO and you can help me with review? Let me know.

Reply:
Maybe add a comment that this property will not be required in future releases?

Reply:
https://github.com/apache/iceberg/pull/9283

Reply:
@agrawalreetika With apache/iceberg#9283 merged, do we still need to set `iceberg.io-impl`?

Reply:
@yingsu00 No after apache/iceberg#9283 and once we upgrade Iceberg in Presto, we don't have to explicitely set `iceberg.io-impl` for Native catalogs. But for `HiveCatalog` we may still need it since Presto has custom FILEIO implementation.

Reply:
Is this waiting a response from @ChunxuTang on this question? https://github.com/prestodb/presto/pull/21399#discussion_r1410244953

Reply:
> Is this waiting a response from @ChunxuTang on this question? https://github.com/prestodb/presto/pull/21399#discussion_r1410244953

I believe not. @agrawalreetika Can you please confirm?

Reply:
No this is good to go for Native catalogs support.